// application-storage/dist/SaveLoadHelper.js
"use strict";
// ----------------------------------------------------------------
// <copyright company="Microsoft Corporation">
//     Copyright (c) Microsoft Corporation.  All rights reserved.
// </copyright>
// ----------------------------------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var SessionContext = tslib_1.__importStar(require("./SessionContext"));
var SaveLocation;
(function (SaveLocation) {
    SaveLocation[SaveLocation["LocalStorage"] = 0] = "LocalStorage";
    SaveLocation[SaveLocation["SessionStorage"] = 1] = "SessionStorage";
})(SaveLocation = exports.SaveLocation || (exports.SaveLocation = {}));
var SaveLoadHelper = /** @class */ (function () {
    function SaveLoadHelper(name, saveLocation) {
        var _this = this;
        this.getValue = function () {
            switch (_this.saveLocation) {
                case SaveLocation.LocalStorage:
                    return SessionContext.loadObjectFromLocalStorage(_this.name);
                case SaveLocation.SessionStorage:
                    return SessionContext.loadObjectFromSessionStorage(_this.name);
            }
        };
        this.setValue = function (value) {
            switch (_this.saveLocation) {
                case SaveLocation.LocalStorage:
                    SessionContext.saveValueToLocalStorage(_this.name, value);
                    break;
                case SaveLocation.SessionStorage:
                    SessionContext.saveValueToSessionStorage(_this.name, value);
                    break;
            }
        };
        this.name = name;
        this.saveLocation = saveLocation;
    }
    return SaveLoadHelper;
}());
exports.SaveLoadHelper = SaveLoadHelper;
exports.default = SaveLoadHelper;
