// application-storage/dist/SessionContext.js
"use strict";
// ----------------------------------------------------------------
// <copyright company="Microsoft Corporation">
//     Copyright (c) Microsoft Corporation.  All rights reserved.
// </copyright>
// ----------------------------------------------------------------
Object.defineProperty(exports, "__esModule", { value: true });
function loadObjectFromSessionStorage(valueName) {
    try {
        var serializedValue = sessionStorage.getItem("" + valueName);
        if (serializedValue === null || serializedValue === undefined || serializedValue === 'undefined') {
            return undefined;
        }
        return JSON.parse(serializedValue);
    }
    catch (err) {
        console.error(err);
        return undefined;
    }
}
exports.loadObjectFromSessionStorage = loadObjectFromSessionStorage;
function loadValueFromSessionStorage(valueName) {
    try {
        var value = sessionStorage.getItem("" + valueName);
        if (value === null || value === undefined || value === 'undefined') {
            return undefined;
        }
        return value;
    }
    catch (err) {
        console.error(err);
        return undefined;
    }
}
exports.loadValueFromSessionStorage = loadValueFromSessionStorage;
function loadObjectFromLocalStorage(valueName) {
    try {
        var serializedValue = localStorage.getItem("" + valueName);
        if (serializedValue === null || serializedValue === undefined || serializedValue === 'undefined') {
            return undefined;
        }
        return JSON.parse(serializedValue);
    }
    catch (err) {
        console.error(err);
        return undefined;
    }
}
exports.loadObjectFromLocalStorage = loadObjectFromLocalStorage;
function loadValueFromLocalStorage(valueName) {
    try {
        var value = sessionStorage.getItem("" + valueName);
        if (value === null || value === undefined || value === 'undefined') {
            return undefined;
        }
        return value;
    }
    catch (err) {
        console.error(err);
        return undefined;
    }
}
exports.loadValueFromLocalStorage = loadValueFromLocalStorage;
function saveValueToLocalStorage(valueName, value) {
    try {
        var serializedState = JSON.stringify(value);
        localStorage.setItem("" + valueName, serializedState);
    }
    catch (err) {
        console.error(err);
    }
}
exports.saveValueToLocalStorage = saveValueToLocalStorage;
function saveValueToSessionStorage(valueName, value) {
    try {
        sessionStorage.setItem("" + valueName, value);
    }
    catch (err) {
        console.error(err);
    }
}
exports.saveValueToSessionStorage = saveValueToSessionStorage;
